acceptance-tests/src/test/scala/io/renku/graph/acceptancetests/knowledgegraph/LineageResourcesSpec.scala

```diff
@@ -0,0 +1,176 @@
+      (lineageData, dataProjects(lineageData.project).generateOne)
+    }
+
+    /** Expected data structure when looking for the grid_plot file
+     *
+     * zhbikes folder clean_data \ / run plan 1 \ bikesParquet plot_data \ / run plan 2 / grid_plot
+     */
+    Scenario("As a user I would like to find a public project's lineage") {
+      Given("some data in the RDF Store")
+      val commitId = commitIds.generateOne
+      mockDataOnGitLabAPIs(project, exemplarData.project.asJsonLD, commitId)
+      `data in the RDF store`(project, commitId)
+
+      When("user calls the lineage endpoint")
+      val response =
+        knowledgeGraphClient GET s"knowledge-graph/projects/${project.path}/files/${exemplarData.`grid_plot entity`.location}/lineage"
+
+      Then("they should get Ok response with project lineage in Json")
+      response.status shouldBe Ok
+      val lineageJson: ACursor = response.jsonBody.hcursor
+
+      lineageJson.downField("edges").as[List[Json]].map(_.toSet) shouldBe theExpectedEdges(exemplarData)
+      lineageJson.downField("nodes").as[List[Json]].map(_.toSet) shouldBe theExpectedNodes(exemplarData)
+
+    }
+  }
+
+  Feature("GET knowledge-graph/projects/<namespace>/<name>/files/<location>/lineage to find a file's lineage") {
+    val user = authUsers.generateOne
+    implicit val accessToken: AccessToken = user.accessToken
+
+    Scenario("As an authenticated user I would like to find lineage of project I am a member of") {
+      val accessibleExemplarData = LineageExemplarData(
+        renkuProjectEntities(fixed(Visibility.Private)).generateOne.copy(
+          path = model.projects.Path("accessible/member-project"),
+          members = Set(personEntities.generateOne.copy(maybeGitLabId = user.id.some))
+        )
+      )
+
+      Given("some data in the RDF Store with a project I am a member of")
+      val commitId = commitIds.generateOne
+      val project  = dataProjects(accessibleExemplarData.project).generateOne
+      mockDataOnGitLabAPIs(project, accessibleExemplarData.project.asJsonLD, commitId)
+      `data in the RDF store`(project, commitId)
+
+      And("I am authenticated")
+      `GET <gitlabApi>/user returning OK`(user)
+
+      When("user fetches the lineage of the project he is a member of")
+
+      val response =
+        knowledgeGraphClient GET (s"knowledge-graph/projects/${project.path}/files/${accessibleExemplarData.`grid_plot entity`.location}/lineage", user.accessToken)
+
+      Then("he should get OK response with project lineage in Json")
+      response.status shouldBe Ok
+
+      val lineageJson: ACursor = response.jsonBody.hcursor
+
+      lineageJson.downField("edges").as[List[Json]].map(_.toSet) shouldBe theExpectedEdges(accessibleExemplarData)
+      lineageJson.downField("nodes").as[List[Json]].map(_.toSet) shouldBe theExpectedNodes(accessibleExemplarData)
+    }
+
+    Scenario("As an unauthenticated user I should not be able to find a lineage from a private project") {
```

Comment:
I think this scenario checks the unauthorized flow as the user token is valid and it's passed to the request